src/javaRosa.js

```diff
@@ -837,6 +838,49 @@ define([
             });
 
         if (options.path === 'labelItext') {
+            if (EXPERIMENTAL_UI) {
+                var at_opts = {
+                    displayTpl: '<li>${name}</li>',
+                    limit: 10,
+                    maxLen: 30,
+                    callbacks: {
+                        matcher: function(flag, subtext) {
+                            var match, regexp;
+                            regexp = new RegExp('(\\s+|^)' + flag + '([\\w_/]*)$', 'gi');
+                            match = regexp.exec(subtext);
+                            return match ? match[2] : null;
+                        }
+                    }
+                },
+                validMugs = _.chain(mug.form.getMugList())
+                             .map(function(mug) {
+                                 return {
+                                     id: mug.ufid,
+                                     name: mug.form.getAbsolutePath(mug, true),
+                                     path: mug.absolutePath,
+                                     type: mug.options.typeName,
+                                 };
+                             })
+                             .filter(function(choice) { return choice.name; })
+                             .value(),
+                dateMugs = _.filter(validMugs, function(mug) {
+                    return mug.type === "Date";
+                });
+                $input.atwho(_.extend(at_opts, {
+                    at: "#form",
+                    data: validMugs,
+                    insertTpl: '<output value="${path}" />',
+                })).atwho(_.extend(at_opts, {
+                    at: "#date",
+                    data: dateMugs,
+                    insertTpl: '<output value="format-date(date(${path}), \'%d/%n/%y\')" />',
+                })).atwho(_.extend(at_opts, {
+                    at: "#fulldate",
```

Comment:
I'm not sure I like adding extra tags like this for things special formatting. It seems the namespace could quickly become cluttered with hundreds of hard-to-remember tags. I think a better idea would be to add formatting extensions once we can represent `<output .../>` as an inline object rather than raw text.